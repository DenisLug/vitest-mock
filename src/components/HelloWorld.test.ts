import { mount } from '@vue/test-utils'
import { expect, vi, test } from 'vitest'
import HelloWorld from './HelloWorld.vue'

vi.mock('./module', () => ({
  default: () => ({
    mockme: () => 'I am mocked'
  })
}))

function createWrapper() {
  return mount(HelloWorld, {
    props: {
      msg: 'Passed Prop Message'
    }
  })
}

test('should match snapshot', () => {
  expect(createWrapper().element).toMatchSnapshot()
})
