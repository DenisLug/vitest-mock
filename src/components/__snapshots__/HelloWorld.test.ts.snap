// Vitest Snapshot v1

exports[`should match snapshot 1`] = `
<div
  data-v-app=""
>
  
  <h1
    data-v-469af010=""
  >
    Passed Prop Message
  </h1>
  <p
    data-v-469af010=""
  >
     Recommended IDE setup: 
    <a
      data-v-469af010=""
      href="https://code.visualstudio.com/"
      target="_blank"
    >
      VS Code
    </a>
     + 
    <a
      data-v-469af010=""
      href="https://github.com/johnsoncodehk/volar"
      target="_blank"
    >
      Volar
    </a>
  </p>
  <p
    data-v-469af010=""
  >
    See 
    <code
      data-v-469af010=""
    >
      README.md
    </code>
     for more information.
  </p>
  <p
    data-v-469af010=""
  >
    <a
      data-v-469af010=""
      href="https://vitejs.dev/guide/features.html"
      target="_blank"
    >
       Vite Docs 
    </a>
     | 
    <a
      data-v-469af010=""
      href="https://v3.vuejs.org/"
      target="_blank"
    >
      Vue 3 Docs
    </a>
  </p>
  <button
    data-v-469af010=""
    type="button"
  >
    count is: 0
  </button>
  <p
    data-v-469af010=""
  >
     Edit 
    <code
      data-v-469af010=""
    >
      components/HelloWorld.vue
    </code>
     to test hot module replacement. 
  </p>
   I am mocked
  
</div>
`;
